pkg/core/mempool/mem_pool.go

```diff
@@ -303,7 +341,7 @@ func (mp *Pool) removeInternal(hash util.Uint256, feer Feer) {
 // RemoveStale filters verified transactions through the given function keeping
 // only the transactions for which it returns a true result. It's used to quickly
 // drop part of the mempool that is now invalid after the block acceptance.
-func (mp *Pool) RemoveStale(isOK func(*transaction.Transaction) bool, feer Feer) {
+func (mp *Pool) RemoveStale(isOK func(*transaction.Transaction, interface{}) bool, feer Feer) {
```

Comment:
I don't think you need this `interface{}`, to make a decision you already have everything needed in the transaction.

Reply:
We need `data`  here because it contains MainTx. And `isOK` for payload includes MainTx checks.

Reply:
You're only using `ValidUntilBlock` (but it's the same as in fallback transaction) and `Hash()` (but it's in the `Conflicts` attribute of fallback).

Reply:
But fallback transaction might have several `Conflicts` attributes (currently we check only the presence of mainTx.Hash in the list of fallbackTx's conflicts attributes). We can either restrict the number of conflicts attributes for the fallback transaction by 1 or reserve position (e.g. first place in the attributes array) for the attribute with the mainTx hash.

Reply:
But if any of them is to enter the chain the fallback can no longer be valid, so you need to check for all of them.

It at the same time shows another potential problem, if I'm to create a fallback conflicting with main tx as well as some $OTHER_TX, I can send this $OTHER_TX right before NVB making the request invalid. Though I'll have to pay for this $OTHER_TX, it can be less than what fallback tx pays with its fees. So we can require fallback to have exactly one conflicting attribute.

Reply:
At least for now, to simplify things. Maybe it's not that big of an issue, but it's safer to add more restrictions at the moment while keeping our main use cases covered.